www/nginx/src/opnsense/service/templates/OPNsense/Nginx/streams.conf

```diff
@@ -100,13 +100,14 @@
 {%   elif server.route_field == 'sni_upstream_map' %}
         proxy_pass $hostmap{{ server.sni_upstream_map.replace('-','') }};
 {%   endif %}
-        proxy_protocol {% if server.proxy_protocol == '1' %}on{% else %}off{% endif %};
+        proxy_protocol {% if upstream.proxy_protocol == '1' %}on{% else %}off{% endif %};
 {%   if server.proxy_responses is defined and server.proxy_responses != '' %}
         proxy_responses {{ server.proxy_responses }};
 {%   endif%}
-{%   if server.trusted_proxies is defined and server.trusted_proxies != '' %}
+{%   if server.trusted_proxies is defined and server.trusted_proxies != '' and server.proxy_protocol is defined and server.proxy_protocol == '1' %}
 {%     for trusted_proxy in server.trusted_proxies.split(',') %}
-        set_real_ip_from {{ trusted_proxy }};
+        # temporarily disabled so as not to create a broken config with a "directive is not allowed" error. need to build with the "with-stream_realip_module" option
+        #set_real_ip_from {{ trusted_proxy }};
```

Comment:
According to https://github.com/opnsense/plugins/pull/3205/files#diff-7fad4db26aa3e57d29d4f7f025a266f134a0a5bd52311263c4eab61c8f64e739R24

@fichtner and @mimugmail  are usually aware how the FreeBSD community works and maintain some packages there. If you need help or they can do it, it would be great as it would also improve the usabillty for non-OPNsense systems too.

Needs to be upstreamed to FreeBSD in the ports tree - OPNsense's fork is here, but this should go to the original:
https://github.com/opnsense/ports/blob/master/www/nginx/Makefile#L181

And then the option needs to be added here:
https://github.com/opnsense/tools/blob/master/config/23.1/make.conf#L107

Reply:
ticket is in progress for FreeBSD port https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=267112

Reply:
@fichtner and @mimugmail , on the advice of @fabianfrz, I kindly ask for your advice here ). 
what is the correct way in this case: try to add the option (`--with-stream_realip_module`) to the opnsense ports repo, wait for the bugzilla ticket to close?
Thanks!

Reply:
If you would ask me I would prefer to implement that ticket without unimplemented placeholders so I would prefer to wait until the patch is upstream.

Reply:
thanks, sorry if I didn’t fully understand: since the `streamserver.trusted_proxies` field is already in the model and template (and I just tried to use it more correctly), probably in this case it makes sense to temporarily hide it from the form (until the `stream_realip_module` module appears in the assembly), so as not to confuse the users?
did I understand correctly?
(hopefully I'll have a chance to try out the patch-Makefile soon and discuss the possibility of using it with @fichtner )